openc3/lib/openc3/topics/limits_event_topic.rb

```diff
@@ -100,11 +135,107 @@ def self.current_set(scope:)
       sets(scope: scope).key('true') || "DEFAULT"
     end
 
-    def self.delete(target_name, packet_name, scope:)
+    # Cleanups up the current_limits and current_limits_settings keys for
+    # a target or target/packet combination
+    def self.delete(target_name, packet_name = nil, scope:)
       limits = Store.hgetall("#{scope}__current_limits")
       limits.each do |item, _limits_state|
-        if item =~ /^#{target_name}__#{packet_name}__/
-          Store.hdel("#{scope}__current_limits", item)
+        if packet_name
+          if item =~ /^#{target_name}__#{packet_name}__/
+            Store.hdel("#{scope}__current_limits", item)
+          end
+        else
+          if item =~ /^#{target_name}__/
+            Store.hdel("#{scope}__current_limits", item)
+          end
+        end
+      end
+
+      limits_settings = Store.hgetall("#{scope}__current_limits_settings")
+      limits_settings.each do |item, _limits_settings|
+        if packet_name
+          if item =~ /^#{target_name}__#{packet_name}__/
+            Store.hdel("#{scope}__current_limits_settings", item)
+          end
+        else
+          if item =~ /^#{target_name}__/
+            Store.hdel("#{scope}__current_limits_settings", item)
+          end
+        end
+      end
+    end
+
+    # Update the local System based on overall state
+    def self.sync_system(scope:)
```

Comment:
How bad is this on a huge DB?

Reply:
Should be fine.  Only changes from the defaults are saved in these variables.